sound/soc/sof/sof-priv.h

```diff
@@ -347,6 +347,8 @@ struct snd_sof_dev {
 
 	/* client devices */
 	struct snd_sof_client *sof_audio;
+	struct snd_sof_client **sof_clients;
+	int sof_num_clients;
```

Comment:
I also don't see why we call out sof_audio explicitly.
it'd be perfectly reasonable to have an implementation without audio?